src/event_pusher/mod_event_pusher_push.erl

```diff
@@ -181,10 +187,56 @@ publish_message(Acc, From, To, Packet) ->
 %%--------------------------------------------------------------------
 %% Helper functions
 %%--------------------------------------------------------------------
+-spec maybe_remove_push_node_from_session_info(jid:jid(), pubsub_node() | undefined) ->
+    ok.
+maybe_remove_push_node_from_session_info(From, undefined) ->
+    %% The node is undefined which means that a user want to disabled all push nodes
+    LUser = From#jid.luser,
+    LServer = From#jid.lserver,
+    AllResources = ejabberd_sm:get_user_present_resources(LUser, LServer),
+    [ejabberd_sm:remove_info(LUser, LServer, Resource, push_notifications) || {_, Resource} <- AllResources],
+    ok;
+maybe_remove_push_node_from_session_info(From, Node) ->
+    LUser = From#jid.luser,
+    LServer = From#jid.lserver,
+    LResource = From#jid.lresource,
+    case my_push_node(LUser, LServer, LResource, Node) of
+        true ->
+            ejabberd_sm:remove_info(LUser, LServer, LResource, push_notifications);
+        false ->
+            find_and_remove_push_node_from_other_session(From, Node)
+    end.
+
+my_push_node(LUser, LServer, LResource, Node) ->
+    {_SUser, _SID, _, SInfo} = ejabberd_sm:get_session(LUser, LServer, LResource),
+    case lists:keyfind(push_notifications, 1, SInfo) of
+        {push_notifications, {Node, _}} ->
+            true;
+        _ ->
+            false
+    end.
+
+find_and_remove_push_node_from_other_session(From, Node) ->
+    LUser = From#jid.luser,
+    LServer = From#jid.lserver,
```

Comment:
Same here, maybe pattern-matching on the function head?

Reply:
Same answer, tried it, line got too long.